core/main/src/firebolt/handlers/provider_registrar.rs

```diff
@@ -119,15 +119,18 @@ impl ProviderRegistrar {
                     });
                 }
             }
-            ProviderResponsePayloadType::ChallengeError => {
+            ProviderResponsePayloadType::ChallengeError
+            | ProviderResponsePayloadType::GenericError => {
                 let external_provider_error: Result<ExternalProviderError, CallError> =
                     params_sequence.next();
-
-                if let Ok(r) = external_provider_error {
-                    return Some(ProviderResponse {
-                        correlation_id: r.correlation_id,
-                        result: ProviderResponsePayload::ChallengeError(r.error),
-                    });
+                match external_provider_error {
+                    Ok(r) => {
+                        return Some(ProviderResponse {
+                            correlation_id: r.correlation_id,
+                            result: ProviderResponsePayload::ChallengeError(r.error),
```

Comment:
Are you sure this is always going to be a `ChallengeError` when the payload type is `ProviderResponsePayloadType::GenericError`?

Reply:
So firebolt open rpc has a generic provider for all error objects which match the ChallengeError 
https://github.com/rdkcentral/firebolt-openrpc/blob/665aad3ece7e8b3e196c1a1bafa34181c646c26e/src/shared/modules.mjs#L878

So i just renamed the challenge Error to generic error now its very scalable

Reply:
Nice, thanks.